awscrt/http.py

```diff
@@ -0,0 +1,153 @@
+# Copyright 2010-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#  http://aws.amazon.com/apache2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+import _aws_crt_python
+from concurrent.futures import Future
+from awscrt.io import ClientBootstrap, TlsConnectionOptions, SocketOptions
+
+
+# Represents an Http connection to a remote endpoint. Everything in this class is non-blocking.
+class HttpClientConnection(object):
+    __slots__ = ('_bootstrap', '_tls_connection_options', '_on_connection_shutdown', '_native_handle')
+
+    # don't call me, I'm private
+    def __init__(self, bootstrap, on_connection_shutdown, tls_connection_options):
+        assert isinstance(bootstrap, ClientBootstrap)
+        assert tls_connection_options is None or isinstance(tls_connection_options, TlsConnectionOptions)
+        assert on_connection_shutdown is not None
+
+        for slot in self.__slots__:
+            setattr(self, slot, None)
+
+        self._bootstrap = bootstrap
+        self._tls_connection_options = tls_connection_options
+        self._on_connection_shutdown = on_connection_shutdown
+        self._native_handle = None
+
+    # Initiates a new connection to host_name and port using socket_options and tls_connection_options if supplied.
```

Comment:
Any reason why we don't follow the PEP for docstrings: https://www.python.org/dev/peps/pep-0287/

Should allow for the best possible user experience in tools